### ISSUE TITLE
Bokeh GridMatrix plots are not well aligned

### ISSUE THREAD

**Issue by maximlt** [holoviz/holoviews#5784]
As seen for instance on this page https://holoviews.org/gallery/demos/bokeh/iris_grouped_grid.html:

<img width="593" alt="image" src="https://github.com/holoviz/holoviews/assets/35924738/82523e31-a403-4646-9021-43dbb87d8dfd">

Bokeh has an example on its docs of a scatter plot matrix (https://docs.bokeh.org/en/latest/docs/user_guide/topics/stats.html#splom) that looks well aligned, hinting at a potential issue in HoloViews directly?

<img width="669" alt="image" src="https://github.com/holoviz/holoviews/assets/35924738/076b773e-9a62-4511-8f0e-0847c1f76b21">

**Comment by Hoxbro** [comment]
On main, it looks even worse, but this is unrelated to the issue described here (and related to having set `min_width=300` as default for elements in https://github.com/holoviz/holoviews/pull/5762. 
![image](https://github.com/holoviz/holoviews/assets/19758978/46d7bf40-bae0-4881-858c-a9e2fc16963b)

**Comment by philippjfr** [comment]
Yeah, that approach to setting `min_height` won't work. It'll have to be done at the figure creation and figure update level and always do something to the effect of `plot.min_height = min(plot.height or float('inf'), 300)`
